x-pack/plugins/fleet/server/services/agent_policies/bump_agent_policies_task.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { v4 as uuidv4 } from 'uuid';
+
+import { appContextService, packagePolicyService } from '..';
+import { runWithCache } from '../epm/packages/cache';
+
+const TASK_TYPE = 'fleet:bump_agent_policies';
+const BATCH_SIZE = 100;
+export function registerBumpAgentPoliciesTask(taskManagerSetup: TaskManagerSetupContract) {
+  taskManagerSetup.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Fleet Bump policies',
+      timeout: '5m',
+      maxAttempts: 3,
+      createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+        let cancelled = false;
+        return {
+          async run() {
+            if (cancelled) {
+              throw new Error('Task has been cancelled');
+            }
+
+            await runWithCache(async () => {
+              await _updatePackagePoliciesThatNeedBump(appContextService.getLogger());
+
+              // TODO agent policies
+            });
+          },
+          async cancel() {
+            cancelled = true;
+          },
+        };
+      },
+    },
+  });
+}
+
+async function getPackagePoliciesToBump() {
+  return await packagePolicyService.list(appContextService.getInternalUserSOClient(), {
```

Comment:
In order to query package policies from all spaces, we need to query with soClient for each space. For this, we need to query all spaces first.

I think similarly the deploy policies task doesn't work correctly, because the logic only queries agent policies from the default space: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/setup/fleet_server_policies_enrollment_keys.ts#L35